### PR TITLE
feat: support etcd prefix as apisix does

### LINKED ISSUE #1456: feat: support etcd prefix as apisix does
# Feature request

## Please describe your feature

apisix has supported etcd prefix to store different clusters' configuration. Dashboard should also support this.
This is very useful in some resource limited situation like dev/test environment.

## Describe the solution you'd like

set multiple prefix in one dashboard, and let the user choose which apisix cluster he want to manage.

## Describe alternatives you've considered

Not yet.

## Additional context

Thank you!

### PR DESCRIPTION
**Pull request by liuyang211**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] New feature provided
- Related issues
Fixes #1456
___
### New feature or improvement
- Describe the details and related test reports.
support etcd prefix as apisix does
___
### Please add the corresponding test cases if necessary.

___
### Backport patches
- Why need to backport?

- Source branch

- Related commits and pull requests

- Target branch

### CODE REVIEW

**Comment by juzhiyuan** [comment]
Hi, I see there have 3 checks failed:

- Backend CLI & Backend E2E: @nic-chen will help you to check what happened.
- Semantic Pull Request: You could ignore safely.

**Comment by starsz** [comment]
Can you write some test cases at "/api/test/shell/cli_test.sh" to test this?

**Comment by juzhiyuan** [comment]
Hi, please sync your codes with the master branch to fix the CI failure.

**Comment by nic-chen** [comment]
@liuyang211  need some test cases. please refer to https://github.com/apache/apisix-dashboard/blob/master/api/test/shell/cli_test.sh#L300-L327
thanks.

File: api/internal/conf/conf.go [hunk 1]
```diff
@@ -225,10 +226,16 @@ func initEtcdConfig(conf Etcd) {
 		endpoints = conf.Endpoints
 	}
 
+	var prefix = "/apisix"
```
Comment (tokers) [inline api/internal/conf/conf.go:229]: Use short assignment is better:

```
prefix := "/apisix"
```

Comment (liuyang211) [reply api/internal/conf/conf.go:229]: OK, I'll change it this way.

**Comment by liuyang211** [comment]
> @liuyang211 need some test cases. please refer to https://github.com/apache/apisix-dashboard/blob/master/api/test/shell/cli_test.sh#L300-L327
> thanks.

I run these cmds to test it locally, 
```shell
cd apisix-dashboard/api
./test/shell/cli_test.sh
```
but fail at
`
./test/shell/cli_test.sh: line 347: ./etcd-v3.4.14-linux-amd64/etcd: No such file or directory
`

Do I miss something?

**Comment by nic-chen** [comment]
@liuyang211 
please have a look at https://github.com/apache/apisix-dashboard/blob/master/.github/workflows/backend-cli-test.yml#L27-L31

If there is something not good, you could fix it. thanks.

**Comment by liuyang211** [comment]
> @liuyang211
> please have a look at https://github.com/apache/apisix-dashboard/blob/master/.github/workflows/backend-cli-test.yml#L27-L31
> 
> If there is something not good, you could fix it. thanks.

Hi, the testcase runs OK in my local machine, but fails in the github workflow, could you review my testcase and give some advice? Thanks.

**Comment by juzhiyuan** [comment]
cc @nic-chen to take a look

File: api/test/shell/cli_test.sh [hunk 2]
```diff
@@ -343,6 +343,75 @@ check_logfile
 ./manager-api stop
```
Comment (nic-chen) [inline api/test/shell/cli_test.sh:343]: need to sleep after stop.

**Comment by nic-chen** [comment]
@liuyang211 
The reason for the failure is that there is no sleep after the previous stop, which causes the Manager API to not really restart successfully:
https://github.com/apache/apisix-dashboard/pull/1477/checks?check_run_id=1946779863#step:5:530

**Review by tokers — APPROVED** [review]

**Comment by liuyang211** [comment]
> @liuyang211
> The reason for the failure is that there is no sleep after the previous stop, which causes the Manager API to not really restart successfully:
> https://github.com/apache/apisix-dashboard/pull/1477/checks?check_run_id=1946779863#step:5:530

Thanks, the testcase is OK now.

**Review by nic-chen — APPROVED** [review]

**Comment by nic-chen** [comment]
@liuyang211 
merged. thanks a lot !

**Comment by liuyang211** [comment]
> @liuyang211
> merged. thanks a lot !

Thank you for your help!
@nic-chen @juzhiyuan @starsz @tokers

**Comment by nic-chen** [comment]
@liuyang211 
More contributions are welcome !  : )

**Pull request merged**